### ISSUE TITLE
a11y

### ISSUE THREAD

**Issue by mowatermelon** [mowatermelon/learn-es6#64]
## 官网

https://addyosmani.com/a11y/

![image](https://user-images.githubusercontent.com/18508817/61133407-fe2cd200-a4ef-11e9-86d5-53453849f754.png)

PhantomJS, which is used for generating the screenshots, is installed automagically, but in some rare cases it might fail to and you'll get an Error: spawn EACCES error. [Download](http://phantomjs.org/download.html) PhantomJS manually and reinstall a11y if that happens.

## 官方仓库

https://github.com/addyosmani/a11y

![image](https://user-images.githubusercontent.com/18508817/61133619-59f75b00-a4f0-11e9-9f0d-26ffb1118ed9.png)

## 官方`npm`地址

> 依赖包地址

https://www.npmjs.com/package/a11y

![image](https://user-images.githubusercontent.com/18508817/61133703-8c08bd00-a4f0-11e9-94cf-8bbd03d8373c.png)

> 在线运行地址

https://npm.runkit.com/a11y

![image](https://user-images.githubusercontent.com/18508817/61134718-c70bf000-a4f2-11e9-9e57-8995fa18e346.png)

## 基础介绍

Easy accessibility audits powered by the Chrome Accessibility Tools

Accessibility 简称（a11y）

## 基础安装

```sh
npm install a11y --save   # npm
yarn add a11y             # Yarn
```

### CLI usage

Run an audit against a URL:

```sh
$ a11y todomvc.com
```

Or multiple URLs:

```sh
$ a11y todomvc.com google.com
```

Also works fine against `localhost`:

```sh
$ a11y localhost:9000
```

And local files:

```sh
$ a11y index.html
```

Even with [glob](https://github.com/isaacs/node-glob#glob) `patterns`:

```sh
$ a11y '**/*.html'
```

> Query help:

```sh
$ a11y --help
```

> Customise viewport size

Type: `string`
Default: `1024x768`

```sh
$ a11y --viewport-size=800x600
```

> Set a custom delay before capturing the page

Type: `number` (seconds)
Default: `1`

```sh
$ a11y --delay=5
```

Useful when the site does things after load that you want to capture.

> Verbose mode:

```sh
$ a11y <url> --verbose
```

Write audit to file:

```sh
$ a11y <url> > audit.txt
```

---

### Module usage

Audit a remote URL and generate an accessibility report:

```javascript
const a11y = require('a11y');

a11y('twitter.com', (err, reports) => {
    const audit = reports.audit; // `a11y` Formatted report
    const report = reports.report; // DevTools Accessibility Audit formatted report
});
```

Work with the output of `reports.audit`:

```javascript
const a11y = require('a11y');

a11y('twitter.com', (err, reports) => {
    for (const report of reports) {
        // Result will be PASS, FAIL or NA
        if (report.result === 'FAIL') {
            // el.heading
            // el.severity
            // el.elements
        }
    }
});
```

> Passing options:

```javascript
const a11y = require('a11y');

const options = {
    viewportSize: '800x600'
};

a11y('twitter.com', options, (err, reports) => {
    // ...
});
```

## 相关网站

- 官方文档，http://momentjs.com/docs/
- 国内版官网，http://momentjs.cn/
